contracts/Executor.sol

```diff
@@ -78,15 +80,31 @@ contract Executor {
                     state.buildInputs(
                         //selector
                         bytes4(command),
-                        bytes32(uint256(indices << 8) | 0xFF)
+                        bytes32(uint256(indices << 8) | NO_VALUE)
                     )
                 );
             } else {
                 revert("Invalid calltype");
             }
 
             require(success, "Call failed");
-            state = state.writeOutputs(bytes1(command << 88), outdata);
+
+            if (
+                flags & COMMAND_TUPLE_RETURN != 0 &&
+                bytes1(command << 88) != bytes1(NO_VALUE)
+            ) {
+                uint8 idx = uint8(bytes1(command << 88));
```

Comment:
Could we put this in its own function? The executor is nice and short right now, and this definitely makes it harder to read.

Reply:
done